tree/dataframe/test/dataframe_merge_results.cxx

```diff
@@ -0,0 +1,61 @@
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RResultPtr.hxx"
+#include "gtest/gtest.h"
+
+TEST(RDataFrameMergeResults, MergeCount)
+{
+    ROOT::RDataFrame df{100};
+    auto mycount = df.Count();
+
+    auto mergeablecount = ROOT::Detail::RDF::GetMergeableValue(mycount);
+    auto othermergeablecount = ROOT::Detail::RDF::GetMergeableValue(mycount);
+
+    mergeablecount->Merge(othermergeablecount);
+    auto mergedcount = mergeablecount->GetValue();
+
+    EXPECT_EQ(mergedcount, 200);
+}
+
+TEST(RDataFrameMergeResults, MergeMean)
+{
+    ROOT::RDataFrame df1{10};
+    ROOT::RDataFrame df2{20};
+    
+    auto col1 = df1.Define("x","double(rdfentry_)");
+    auto col2 = df2.Define("x","double(rdfentry_)");
+
+    auto mean1 = col1.Mean("x");
+    auto mean2 = col2.Mean("x");
+
+    auto mergeablemean = ROOT::Detail::RDF::GetMergeableValue(mean1);
+    auto othermergeablemean = ROOT::Detail::RDF::GetMergeableValue(mean2);
+
+    mergeablemean->Merge(othermergeablemean);
+
+    auto mergedmean = mergeablemean->GetValue();
+
+    EXPECT_NEAR(mergedmean, 7.83333333333, 0.01);
+}
+
+TEST(RDataFrameMergeResults, MergeHist)
+{
+    ROOT::RDataFrame df1{100};
+    ROOT::RDataFrame df2{100};
+    
+    auto col1 = df1.Define("x","double(rdfentry_)");
+    auto col2 = df2.Define("x","double(rdfentry_)");
+
+    auto hist1 = col1.Histo1D("x");
+    auto hist2 = col2.Histo1D("x");
+
+    auto mergeablehist = ROOT::Detail::RDF::GetMergeableValue(hist1);
+    auto othermergeablehist = ROOT::Detail::RDF::GetMergeableValue(hist2);
+
+    mergeablehist->Merge(othermergeablehist);
+
+    auto &mergedhist = mergeablehist->GetValue();
+
+    EXPECT_EQ(mergedhist.GetEntries(), 200);
+    EXPECT_NEAR(mergedhist.GetMean(), 50, 0.5);
+}
```

Comment:
please add a test in which you get two mergeable results of the same type but from different actions and you try to merge those. an exception should be thrown in this case (no way to find it out at compile time, but we should have a way to find it out a runtime).

Reply:
I'm not sure I understand "of the same type but from different actions". Do you mean e.g. a result from a `Max` and from a `Min` that both hold  a `double` as a result?

Reply:
> Do you mean e.g. a result from a Max and from a Min that both hold a double as a result?

yep

Reply:
Added in the latest commit